test/parallel/test-ttywrap-stack.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that console.log
+// will not crash the process if there
+// is not enough space on the V8 stack
+
+const done = common.mustCall(() => {});
```

Comment:
This should be unnecessary when common is required.

Reply:
Please correct me if I'm wrong, but the promise will be rejected anyway, so if we don't catch the error, the test will fail. Despite the `try..catch`, an error is thrown from the `console.log` call in the `catch` block again (when constructing function template in C++), and the promise gets rejected.

Reply:
You are correct, in that case we should use `assert.rejects` instead.

Reply:
Tried using

```js
assert.rejects(async () => {
  try {
    await test();
  } catch (err) {
    console.log(err);
  }
});
```

but it triggers `crashOnUnhandledRejection` from `common.js` and the test fails. My guess is that `assert.rejects` throws an error itself due to low stack space.

Reply:
hmm, I guess this test is fine as it is then, we can't really use our normal test utilities in this circumstance.